src/Elements/types.jl

```diff
@@ -0,0 +1,180 @@
+    for child in children
+        if isa(child, Tuple)
+            childconf = child[end]
+            isa(childconf, ConfigNode) ||
+                    throw(ArgumentError("`page` content `Tuple`: Last item must be a `config`."))
+            [push!(contentchildren, content(child[i], childconf)) for i in 1:length(child)-1]
+        else
+            push!(contentchildren, content(child, PreConfig()))
+        end
+    end
+    this = isempty(conf.cargs)                                                              ?
+            Page(name, PreConfig(), meta, NullSection(), contentchildren) :
+            Page(name, conf, meta, NullSection(), contentchildren)
+    return this
+end
+
+
+## Content nodes
+type Content <: ContentNode
+    typename    :: Symbol
+    config      :: ConfigNode
+    parent      :: PageNode
+    data        :: Any
+end
+
+function content(child, conf::ConfigNode)
+    if isa(child, AbstractString)
+        headertype = getheadertype(child)
+        if headertype == :none
+            filename = abspath(child)
+            Content(:text, conf, NullPage(), isfile(filename) ? readall(filename) : child)
+        else
+            Content(headertype, conf, NullPage(), child)
+        end
+    elseif isa(child, Module)
+        Content(:module, conf, NullPage(), child)
+    else
+        throw(ArgumentError("`$(typeof(child))` is not a valid `content` type."))
+    end
+end
+
+
+## Related functions
+# sets: parents, final configuration
+function postprocess!(parent, children::Vector)
+    for child in children
+        child.parent = parent
+        child.config = isempty(child.config.cargs)  ?
+                            deepcopy(parent.config) :
+                            update_config!(deepcopy(parent.config), deepcopy(child.config.cargs))
+        if isa(child, Union(Section, Page))
+            isempty(child.children) || postprocess!(child, child.children)
+        # these have no field children: but valid just skip them
+        elseif !isa(child, ContentNode)
+            throw(ArgumentError("`$(typeof(child))` is not a valid node structure type."))
+        end
+    end
+end
+
+# get the method name for a node structure object
+function getnodename(nodeobj)
+    isa(nodeobj, Document)  && return "document"
+    isa(nodeobj, Section)   && return "section"
+    isa(nodeobj, Page)      && return "page"
+    throw(ArgumentError("`$(typeof(nodeobj))` is not a valid node structure type."))
+end
+
+function getheadertype(s::AbstractString)
+    headertype = :none
+    if s[1] == '#'
+        startswith(s, "# ")         && (headertype = :header1)
+        startswith(s, "## ")        && (headertype = :header2)
+        startswith(s, "### ")       && (headertype = :header3)
+        startswith(s, "#### ")      && (headertype = :header4)
+        startswith(s, "##### ")     && (headertype = :header5)
+        startswith(s, "###### ")    && (headertype = :header6)
+    end
+    return headertype
+end
```

Comment:
Would counting the number of ``#``, with a regex or something, and then constructing a ``:header<n>`` work here to reduce the repetition?